packages/databyss-services/pages/PageProvider.tsx

```diff
@@ -60,8 +61,30 @@ const PageProvider: React.FunctionComponent<PropsType> = ({
     return null
   }
 
+  const registerBlockRef = (id: string, ref: React.Ref<HTMLInputElement>) => {
+    dispatch(registerRef(id, ref))
```

Comment:
I think we should maintain the refDict as a "class variable" (ref) in PageProvider (i.e. outside of the reducer state), as they are low-level UI references and don't need dispatch/listener features. Let's discuss though.